### PR TITLE
Donation block: Add block settings

### PR DESCRIPTION
**Pull request by mmtr**
#### Changes proposed in this Pull Request

Add a settings panel to the Donation block so user can toggle monthly, annual, and “Other” amount options on or off.

Jun-30-2020 16-00-51

#### Testing instructions

* Generate a new FSE build and sync it to your sandbox: `cd apps/full-site-editing; yarn dev --sync`.
* Sandbox the API and the store. More info in PCYsg-lW4-p2 #sandbox-method.
* Sandbox a test site with a paid plan, and ensure the site has the `fse-donations-block` blog sticker applied (if not, it can be done from the site's Blog RC).
* Insert a Donations block in a new post with the block editor.
* Verify you can toggle all the settings in the right sidebar and the UI updates accordingly.
* Save the post and reload the editor.
* Make sure the block loads with the previous state.
* Switch to the code editor and confirm the attributes are correctly serialized. You should get something like this if all settings are enabled:
```
<!-- wp:a8c/donations {"oneTimePlanId":573,"monthlyPlanId":569,"annuallyPlanId":570,"showCustomAmount":true} -->
<div class="wp-block-a8c-donations">Donations block</div>
<!-- /wp:a8c/donations -->
```

Part of #42856

### CODE REVIEW

File: apps/full-site-editing/full-site-editing-plugin/donations/donations.js [hunk 1]
```diff
@@ -6,53 +6,106 @@ import classNames from 'classnames';
...
+								onClick={ () => setActiveTab( 'one-time' ) }
+							>
+								{ __( 'One-Time', 'full-site-editing' ) }
+							</Button>
+							{ monthlyPlanId && (
+								<Button
+									className={ classNames( { 'is-active': isTabActive( 'monthly' ) } ) }
+									onClick={ () => setActiveTab( 'monthly' ) }
+								>
+									{ __( 'Monthly', 'full-site-editing' ) }
```
Comment (a8ci18n) [inline apps/full-site-editing/full-site-editing-plugin/donations/donations.js:49]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **69** times:
`translate( 'Monthly' )` *ES Score: 13*

File: apps/full-site-editing/full-site-editing-plugin/donations/donations.js [hunk 2]
```diff
@@ -6,53 +6,106 @@ import classNames from 'classnames';
...
+								</Button>
+							) }
+							{ annuallyPlanId && (
+								<Button
+									className={ classNames( { 'is-active': isTabActive( 'annually' ) } ) }
+									onClick={ () => setActiveTab( 'annually' ) }
+								>
+									{ __( 'Annually', 'full-site-editing' ) }
```
Comment (a8ci18n) [inline apps/full-site-editing/full-site-editing-plugin/donations/donations.js:57]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'billed annually' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/donation-block-settings)

File: apps/full-site-editing/full-site-editing-plugin/donations/donations.js [hunk 3]
```diff
@@ -6,53 +6,106 @@ import classNames from 'classnames';
...
+						</div>
+						{ showCustomAmount && (
+							<div>{ __( 'Or enter a custom amount', 'full-site-editing' ) } </div>
+						) }
+					</div>
 				</div>
-			</div>
-		</div>
+				<InspectorControls>
+					<PanelBody title={ __( 'Settings', 'full-site-editing' ) }>
```
Comment (a8ci18n) [inline apps/full-site-editing/full-site-editing-plugin/donations/donations.js:74]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **42** times:
`translate( 'Settings', { context: 'NewDash Page Title'} )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/donation-block-settings)

File: apps/full-site-editing/full-site-editing-plugin/donations/donations.js [hunk 4]
```diff
@@ -6,53 +6,106 @@ import classNames from 'classnames';
...
-					className={ classNames( 'donations__tab', { active: isActive( 'monthly' ) } ) }
-				>
-					{ __( 'Monthly', 'full-site-editing' ) }
-				</div>
-				<div
-					id="donations__tab-annually"
-					className={ classNames( 'donations__tab', { active: isActive( 'annually' ) } ) }
-				>
-					{ __( 'Annually', 'full-site-editing' ) }
+		<>
```
Comment (kwight) [inline apps/full-site-editing/full-site-editing-plugin/donations/donations.js:32]: We shouldn't need the fragment any more, now with the `<Block.div>`.

File: apps/full-site-editing/full-site-editing-plugin/donations/donations.js [hunk 5]
```diff
@@ -6,53 +6,106 @@ import classNames from 'classnames';
...
+const Donations = ( { attributes, products, setAttributes, siteSlug, stripeConnectUrl } ) => {
+	const { oneTimePlanId, monthlyPlanId, annuallyPlanId, showCustomAmount } = attributes;
 	const [ activeTab, setActiveTab ] = useState( 'one-time' );
 
-	const isActive = ( button ) => ( activeTab === button ? 'active' : null );
+	const isTabActive = ( tab ) => activeTab === tab;
+
+	useEffect( () => {
+		if ( ! oneTimePlanId ) {
+			setAttributes( { oneTimePlanId: products[ 'one-time' ] } );
```
Comment (gwwar) [inline apps/full-site-editing/full-site-editing-plugin/donations/donations.js:27]: Let's start with this but I'm not sure if we want to start with one interval type, or all 3.

File: apps/full-site-editing/full-site-editing-plugin/donations/donations.js (hunk 4 above)
Comment (gwwar) [reply apps/full-site-editing/full-site-editing-plugin/donations/donations.js]: Good eye @kwight, I've updated this in https://github.com/Automattic/wp-calypso/pull/43798/commits/4e11be462db68bea1c655ea0e5de0547653830b1

File: apps/full-site-editing/full-site-editing-plugin/donations/donations.js [hunk 6]
```diff
@@ -6,53 +6,104 @@ import classNames from 'classnames';
...
+							onClick={ () => setActiveTab( 'one-time' ) }
+						>
+							{ __( 'One-Time', 'full-site-editing' ) }
+						</Button>
+						{ monthlyPlanId && (
+							<Button
+								className={ classNames( { 'is-active': isTabActive( 'monthly' ) } ) }
+								onClick={ () => setActiveTab( 'monthly' ) }
+							>
+								{ __( 'Monthly', 'full-site-editing' ) }
```
Comment (a8ci18n) [inline apps/full-site-editing/full-site-editing-plugin/donations/donations.js:48]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **69** times:
`translate( 'Monthly' )` *ES Score: 13*

File: apps/full-site-editing/full-site-editing-plugin/donations/donations.js [hunk 7]
```diff
@@ -6,53 +6,104 @@ import classNames from 'classnames';
...
+							</Button>
+						) }
+						{ annuallyPlanId && (
+							<Button
+								className={ classNames( { 'is-active': isTabActive( 'annually' ) } ) }
+								onClick={ () => setActiveTab( 'annually' ) }
+							>
+								{ __( 'Annually', 'full-site-editing' ) }
```
Comment (a8ci18n) [inline apps/full-site-editing/full-site-editing-plugin/donations/donations.js:56]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **18** times:
`translate( 'billed annually' )` *ES Score: 7*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/donation-block-settings)

File: apps/full-site-editing/full-site-editing-plugin/donations/donations.js [hunk 8]
```diff
@@ -6,53 +6,104 @@ import classNames from 'classnames';
...
+						{ isTabActive( 'annually' ) && __( 'Make a yearly donation', 'full-site-editing' ) }
+					</div>
+					{ showCustomAmount && (
+						<div>{ __( 'Or enter a custom amount', 'full-site-editing' ) } </div>
+					) }
 				</div>
 			</div>
-		</div>
+			<InspectorControls>
+				<PanelBody title={ __( 'Settings', 'full-site-editing' ) }>
```
Comment (a8ci18n) [inline apps/full-site-editing/full-site-editing-plugin/donations/donations.js:73]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **42** times:
`translate( 'Settings', { context: 'NewDash Page Title'} )` *ES Score: 8*
See 2 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/donation-block-settings)

**Pull request merged**